Error in executing the example in ocaml 4.06

Tried this in a file using jbuilder.

```

open Depyt

type t = Foo | Bar of string option;;
let t =
    variant "v" (fun foo bar -> function Foo -> foo | Bar x -> bar)
    |~ case0 "Foo" Foo
    |~ case1 "Bar" (option string) (fun x -> Bar x)
    |> sealv

Fmt.pr "t = %a\n" (pp t) Foo;;

compare t Foo (Bar (Some "a"));;


```

The compile says the following errror.

```
jbuilder build bin/revolt.exe
      ocamlc bin/.revolt.eobjs/revolt.{cmi,cmo,cmt} (exit 2)
(cd _build/default && /usr/bin/ocamlc.opt -w -40 -w +A-40-42 -g -bin-annot -I bin/.revolt.eobjs -I /home/ram/.opam/default/lib/bytes -I /home/ram/.opam/default/lib/cmdliner -I /home/ram/.opam/default/lib/cstruct -I /home/ram/.opam/default/lib/depyt -I /home/ram/.opam/default/lib/fmt -I /home/ram/.opam/default/lib/jsonm -I /home/ram/.opam/default/lib/ocplib-endian -I /home/ram/.opam/default/lib/result -I /home/ram/.opam/default/lib/sexplib -I /home/ram/.opam/default/lib/sexplib/0 -I /home/ram/.opam/default/lib/uchar -I /home/ram/.opam/default/lib/uutf -no-alias-deps -o bin/.revolt.eobjs/revolt.cmo -c -impl bin/revolt.ml)
File "bin/revolt.ml", line 9, characters 7-51:
Error: This expression has type
         (t, string option -> t Depyt.case_p) Depyt.case
       but an expression was expected of type (t, t Depyt.case_p) Depyt.case
       Type string option -> t Depyt.case_p is not compatible with type
         t Depyt.case_p 
make: *** [Makefile:11: build] Error 1


```

The reason i tried this was, i building a project using irimin constructs. 

I see the [module Type](https://github.com/mirage/irmin/blob/master/src/irmin/irmin.mli) doing the same job like `Depyt`. 

1. Why was this library - `Depyt` not used and reinvented again in irmin ? 
2. How do i fix this error, or should i go with  [module Type](https://github.com/mirage/irmin/blob/master/src/irmin/irmin.mli).

--- a/.gitignore
+++ b/.gitignore
@@ -6,3 +6,4 @@ tmp
 *.install
 *.native
 *.byte
+_opam

--- a/README.md
+++ b/README.md
@@ -23,22 +23,22 @@ between intermediate formats.
 For instance, to define variants:
 
 ```ocaml
+# #require "depyt";;
 # open Depyt;;
 # type t = Foo | Bar of string option;;
+type t = Foo | Bar of string option
 # let t =
-    variant "v" (fun foo bar -> function Foo -> foo | Bar x -> bar)
+    variant "v" (fun foo bar -> function Foo -> foo | Bar x -> bar x)
     |~ case0 "Foo" Foo
     |~ case1 "Bar" (option string) (fun x -> Bar x)
     |> sealv
+    ;;
 val t : t Depyt.t = <abstr>
-
-# Fmt.pr "t = %a\n" (pp t) Foo;;
+# Fmt.pr "t = %a\n%!" (dump t) Foo;;
 t = Foo
 - : unit = ()
-
 # compare t Foo (Bar (Some "a"));;
 - : int = -1
-
 # compare t Foo (Bar (Some "a"));;
 - : int = -1
 ```
@@ -48,21 +48,20 @@ t = Foo
 To define records:
 
 ```ocaml
-# open Depyt;;
 # type t = { foo: int option; bar: string list };;
+type t = { foo : int option; bar : string list; }
 # let t =
     record "r" (fun foo bar -> { foo; bar })
     |+ field "foo" (option int) (fun t -> t.foo)
     |+ field "bar" (list string) (fun t -> t.bar)
     |> sealr
-va t : t Depyt.t = <abstr>
-
-# Fmt.pr "%a\n" (pp t) { foo = Som 3; bar = ["foo"] };;
+    ;;
+val t : t Depyt.t = <abstr>
+# Fmt.pr "%a\n%!" (dump t) { foo = Some 3; bar = ["foo"] };;
 { foo = Some 3; bar = ["foo"]; }
 - : unit = ()
-
-(* [None] fields do not appear in the generated JSON *)
-# Fmt.pr "%a\n" (pp_json t) { foo = None; bar = ["1";"2"] };;
+# (* [None] fields do not appear in the generated JSON *)
+# Fmt.pr "%a\n%!" (pp_json t) { foo = None; bar = ["1";"2"] };;
 {"bar":["1","2"]}
 - : unit = ()
 ```

--- a/test/jbuild
+++ b/test/jbuild
@@ -8,3 +8,10 @@
   ((name runtest)
    (deps (test.exe))
    (action (run ${^} --color=always))))
+
+; (alias
+; ((name runtest)
+;  (deps (../README.md (package depyt)))
+;  (action (progn
+;           (run mdx test ${<})
+;           (diff? ${<} ${<}.corrected)))))

